service/matching/matchingEngine.go

```diff
@@ -1185,6 +1185,9 @@ func (e *matchingEngineImpl) redirectToVersionedQueueForAdd(
 	if kind == enumspb.TASK_QUEUE_KIND_STICKY {
 		return taskQueue, nil
 	}
+	if !stamp.GetUseVersioning() {
+		return taskQueue, nil
+	}
```

Comment:
don't do this here, instead, leave the build stamp out of the AddWorkflowTaskRequest on the history side (there's no reason for matching to know about it if it's not going to use it)

Reply:
I think we'll need to to not send the stamp directly from history as I mentioned in my review of your last PR.
For now I'm leaving this on so the tests pass.

Reply:
Let's fix before we merge and this of course depends on your changes too.